src/com/amazon/corretto/crypto/provider/ConstantTime.java

```diff
@@ -0,0 +1,69 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazon.corretto.crypto.provider;
+
+/**
+ * Contains several constant time utilities
+ */
+final class ConstantTime {
+    private ConstantTime() {
+        // Prevent instantiation
+    }
+
+    /**
+     * Equivalent to {@code val != 0 ? 1 : 0}
+     */
+    static final int isNonZero(int val) {
+        return ((val | -val) >>> 31) & 0x01; // Unsigned bitshift
+    }
+
+    /**
+     * Equivalent to {@code val == 0 ? 1 : 0}
+     */
+    static final int isZero(int val) {
+         return 1 - isNonZero(val);
+    }
+
+    /**
+     * Equivalent to {@code val < 0 ? 1 : 0}
+     */
+    static final int isNegative(int val) {
+        return (val >>> 31) & 0x01;
+    }
+
+    /**
+     * Equivalent to {@code x == y ? 1 : 0}
+     */
+    static final int equal(int x, int y) {
+        final int difference = x - y;
```

Comment:
~~Note: Can we make sure we're testing for cases where `(x - y) >=  4294967295`? You might have to go out of your way to trigger an underflow to `0` (if it's even possible) but we should make sure we're covering this case.~~

Confirmed that the tests are covering this condition. Given that the tests are not failing, I am happy with this implementation.